src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -0,0 +1,469 @@
+    return client.list(resource_group_name, registry_name)
+
+
+def acr_task_delete(cmd,
+                    client,
+                    task_name,
+                    registry_name,
+                    resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+    return client.delete(resource_group_name, registry_name, task_name)
+
+
+def acr_task_update(cmd,  # pylint: disable=too-many-locals
+                    client,
+                    task_name,
+                    registry_name,
+                    resource_group_name=None,
+                    # task parameters
+                    status=None,
+                    os_type=None,
+                    cpu=None,
+                    timeout=None,
+                    context_path=None,
+                    commit_trigger_enabled=None,
+                    git_access_token=None,
+                    branch=None,
+                    image_names=None,
+                    no_push=None,
+                    no_cache=None,
+                    file=None,
+                    values=None,
+                    arg=None,
+                    secret_arg=None,
+                    set_value=None,
+                    set_secret=None,
+                    base_image_trigger_enabled=None,
+                    base_image_trigger_type=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    task = client.get(resource_group_name, registry_name, task_name)
+    step = task.step
+    if isinstance(step, DockerBuildStep):
+        if file and file.endswith(ALLOWED_TASK_FILE_TYPES):
+            raise CLIError("File for docker build step has an invalid suffix: {}."
+                           " The following suffixes are not allowed: {}"
+                           .format(file, ALLOWED_TASK_FILE_TYPES))
+        if arg is None and secret_arg is None:
+            arguments = None
+        else:
+            arguments = (arg if arg else []) + (secret_arg if secret_arg else [])
+        step = DockerBuildStepUpdateParameters(
+            image_names=image_names,
+            is_push_enabled=not no_push,
+            no_cache=no_cache,
+            docker_file_path=file,
+            arguments=arguments,
+            context_path=context_path
+        )
+    elif isinstance(step, FileTaskStep):
+        if file and not file.endswith(ALLOWED_TASK_FILE_TYPES):
+            raise CLIError("File for task build step has an invalid suffix: {}."
+                           " It must have one of the following suffixes: {}"
+                           .format(file, ALLOWED_TASK_FILE_TYPES))
+        if set_value is None and set_secret is None:
+            values = None
+        else:
+            values = (set_value if set_value else []) + (set_secret if set_secret else [])
```

Comment:
NIT: this if-else block can be moved out of the outer if-else so it won't be duplicated.